src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md

```diff
@@ -1,4 +1,400 @@
+this would be a URL.
+
+Finally we can use the extension in a test script:
+
+<CodeGroup labels={["test.js"]} lineNumbers={[true]}>
+
+```go
+import compare from 'k6/x/compare';
+
+export default function () {
+  console.log(compare.isGreater(2, 1));
+}
+```
+
+</CodeGroup>
+
+And run the test with `./k6 run test.js`, which should output `INFO[0000] true`.
+
+
+#### Additional features
+
+The k6 Go-JS bridge has a few features we should highlight:
+
+- Go method names will be converted from Pascal case to Camel case when
+  accessed in JS, as in the example above: `IsGreater` becomes `isGreater`.
+
+<!-- FIXME: Markdown magic to escape backticks in Go struct tags? <pre> doesn't work... -->
+- Similarly, Go field names will be converted from Pascal case to Snake case.
+  For example, the struct field `SomeField string` is accessible in JS
+  as the `some_field` object property. This behavior is configurable with the
+  `js` struct tag, so this can be changed with `SomeField string ``js:"someField"```
+  or the field can be hidden with `js:"-"`.
+
+- Method names prefixed with `X` are interpreted as constructors in JS,
+  and will support the `new` operator.
+  For example, defining the following method on the above struct:
+
+<CodeGroup labels={["compare.go"]} lineNumbers={[false]}>
+
+```go
+func (*Compare) XComparator() *Comparator {
+	return &Comparator{}
+}
+```
+
+</CodeGroup>
+
+  Would allow creating a `Comparator` instance in JS with `new compare.Comparator()`,
+  which is a bit more idiomatic to JS.
+
+- Methods that specify `context.Context` or `*context.Context` as the first
+  argument will be passed the `Context` instance used internally in k6,
+  which has attached some useful objects for inspecting the internal execution
+  state, such as
+  [`lib.State`](https://github.com/grafana/k6/blob/v0.33.0/lib/state.go#L43)
+  or VU state, [`lib.ExecutionState`](https://github.com/grafana/k6/blob/v0.33.0/lib/execution.go#L142),
+  and the [`goja.Runtime`](https://github.com/dop251/goja/blob/705acef95ba3654f89c969d9e792ac5f49215350/runtime.go#L162) instance
+  the VU is using to execute the script.
+  This feature is used extensively in the
+  [`xk6-execution`](https://github.com/grafana/xk6-execution) extension.
+
+<!-- TODO: Mention common.Bind() here? HasModuleInstancePerVU? -->
+
+
+#### Things to keep in mind
+
+- The code in the `default` function (or another function specified by
+  [`exec`](/using-k6/scenarios/#common-options)) will be executed many
+  times during a test run and possibly in parallel by thousands of VUs.
+  As such any operation of your extension meant to run in that context
+  needs to be performant and [thread-safe](https://en.wikipedia.org/wiki/Thread_safety).
+- Any heavy initialization should be done in the [`init`
+  context](/javascript-api/init-context/)...
+- Custom metric emission can be done by creating new metrics using
+  [`stats.New()`](https://github.com/grafana/k6/blob/v0.33.0/stats/stats.go#L449)
```

Comment:
```suggestion
  [stats.New()](https://github.com/grafana/k6/blob/v0.33.0/stats/stats.go#L449)
```

Reply:
I see your point that links on preformatted text are invisible without hovering over them, but that's a UI problem, not a semantic one :)

So I'd rather leave this and other usage as is, and create an issue for someone from frontend (PixelPoint?) to take a look at fixing the rendering. @ppcano WDYT?